src/QueryFetcher.ts

```diff
@@ -116,12 +117,14 @@ export class QueryFetcher<TOperationType extends OperationType = OperationType>
 
         const { onComplete, onResponse } = options;
         let fetchHasReturned = false;
-        const onNext = (_o: OperationDescriptor, snapshot: Snapshot): void => {
+        const onNext = (_o: OperationDescriptor, snapshot: Snapshot, response: GraphQLResponse): void => {
             if (!this.snapshot) {
                 this.snapshot = snapshot;
                 this.subscribe(snapshot);
                 this.resolveResult();
-                if (fetchHasReturned) {
+                const responses = Array.isArray(response) ? response : [response];
+                const isFinal = responses.some((x) => x != null && x.extensions?.is_final === true);
```

Comment:
deepening the proposed solution for defer https://github.com/facebook/relay/pull/3121#issue-652496136

prefer to change the logic using hasNext (spec-complaint) instead of is_final (FB specific):

```ts
const isIncremental = responses.some((x) => x != null && x.hasNext === true);
if (fetchHasReturned && isIncremental) {
    this.forceUpdate();
}
```



In this way, no code modification is needed to optimize the render in case an incremental query is not used but it could be a breaking change for those using it with other specifications